util/topgen/validate.py

```diff
@@ -293,6 +293,15 @@ def check_clocks_resets(top, ipobjs, ip_idxs, xbarobjs, xbar_idxs):
 
     error = 0
 
+    # there should only be one each of pwrmgr/clkmgr/rstmgr
+    pwrmgrs = [m for m in top['module'] if m['type'] == 'pwrmgr']
+    clkmgrs = [m for m in top['module'] if m['type'] == 'clkmgr']
+    rstmgrs = [m for m in top['module'] if m['type'] == 'rstmgr']
+
+    if len(pwrmgrs) == 1 and len(clkmgrs) == 1 and len(rstmgrs) is False:
```

Comment:
I first confused this. So the expression is:

`(len(pwrmgrs) == 1) and (len(clkmgrs) == 1) and len(rstmgrs)) is False` ? should it be `len(rstmgr) == 1`?
or, `len(pwrmgr) * len(clkmgr) * len(rstmgr) != 1` may work.

Reply:
ooo i like the second one. Let me do that